pkg/compare/parsing.go

```diff
@@ -101,24 +178,71 @@ func (r *Reference) getMissingCRs(matchedTemplates map[string]bool) (map[string]
 	return crs, count
 }
 
-var defaultFieldsToOmit = [][]string{{"metadata", "uid"},
-	{"metadata", "resourceVersion"},
-	{"metadata", "generation"},
-	{"metadata", "generateName"},
-	{"metadata", "creationTimestamp"},
-	{"metadata", "finalizers"},
-	{"kubectl.kubernetes.io/last-applied-configuration"},
-	{"metadata", "annotations", "kubectl.kubernetes.io/last-applied-configuration"},
-	{"status"},
+const builtInPathsKey = "cluster-compare-built-in"
+
+var builtInPaths = []string{
+	"metadata.resourceVersion",
+	"metadata.generation",
+	"metadata.uid",
+	"metadata.generateName",
+	"metadata.creationTimestamp",
+	"metadata.finalizers",
+	`"kubectl.kubernetes.io/last-applied-configuration"`,
+	`metadata.annotations."kubectl.kubernetes.io/last-applied-configuration"`,
+	"status",
+}
+
+type Path struct {
```

Comment:
This part is a declaration of a new "language",  like jsonPath, just that for now the implementation does not include many features so it may look not that big of a deal.

I don't think its a good idea to implement a custom syntax for this path,    its better to use some thing that already exits (we can look in to it together ).

Also if we do choose implementing are own syntax, this should be placed in a different package and well documented. also i think the implementation should be changed to be more extendable, for example if we are going to support regex this will need refactoring, and will make it more complex to understand. I think in this case there are neater ways to define parsed languages in go lang (quick search: https://github.com/alecthomas/participle probaly there are better then that also)
    
I do strongly think its smart to look for a known approach that  is already implemented, and invest the time in finding one. also possible to take a known solution and  remove features from it or constraint if its to complex.
  I think this also connect to are discussion when we wanted to add comments to the templates, where also the outcome was not to create new formats.

Reply:
@pixelsoccupied what do you think?

Reply:
update: take a look at the first answer here:
https://stackoverflow.com/questions/47489745/splitting-a-string-at-space-except-inside-quotation-marks
i guess also an option is to simplify it to be a like csv with a dot separator untill think about the regexes

Reply:
If it starts getting more complex than the regex case (e.g. logic or more extensive operators) perhaps we should go down the route of a properly defined language with a parser and lexer but ATM its overkill. ATM its convenient way of describing a list of segments (even with regex) and I think it should stay that way.

I went down the path of using a jq lib but due to the complexity of jq it wasn't easy to do the back tracking, to clean up the empty structures. 
Thinking about it perhaps we could instead of deleting the item, we could replace it with a marker that we can then search for and delete, then do the back tracking. I can try this if you would like.

Its not very hard to extend it to regex can just denote regex with something like backticks. https://github.com/nocturnalastro/kube-compare/pull/2/files#diff-a82c27d824a8a061c8c740353caf25344e333a1a4d77c6c2e2d57824a5cb31b1R209
I actually initially wrote it with the regex case in but thought it needed some refinement so I split it off. 

There is other ways to approach parsing the line though that might be clearer I'll give some of them a go for comparison.

Reply:
Given that we only need prefix and not full regex support we decided to go with csv reader to split and then just a flag for isPrefix